prism/src/main/java/org/apache/falcon/service/BacklogMetricEmitterService.java

```diff
@@ -0,0 +1,313 @@
+
+    private static BacklogMetricStore backlogMetricStore = new BacklogMetricStore();
+
+    private static final BacklogMetricEmitterService SERVICE = new BacklogMetricEmitterService();
+
+    private static MetricNotificationService metricNotificationService =
+            Services.get().getService(MetricNotificationService.SERVICE_NAME);
+
+    public static BacklogMetricEmitterService get() {
+        return SERVICE;
+    }
+
+    private BacklogMetricEmitterService() {
+    }
+
+    private ScheduledThreadPoolExecutor scheduledThreadPoolExecutor1 = new ScheduledThreadPoolExecutor(1);
+    private ScheduledThreadPoolExecutor scheduledThreadPoolExecutor2 = new ScheduledThreadPoolExecutor(1);
+
+
+    public static final ThreadLocal<SimpleDateFormat> DATE_FORMAT = new ThreadLocal<SimpleDateFormat>() {
+        @Override
+        protected SimpleDateFormat initialValue() {
+            SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH-mm'Z'");
+            format.setTimeZone(TimeZone.getTimeZone("UTC"));
+            return format;
+        }
+    };
+
+    private static ConcurrentHashMap<Entity, List<MetricInfo>> entityBacklogs = new ConcurrentHashMap<>();
+
+    @Override
+    public void highSLAMissed(String entityName, EntityType entityType, String clusterName,
+                              Date nominalTime) throws FalconException {
+        if (entityType != EntityType.PROCESS) {
+            return;
+        }
+        Entity entity = EntityUtil.getEntity(entityType, entityName);
+        entityBacklogs.putIfAbsent(entity, Collections.synchronizedList(new ArrayList<MetricInfo>()));
+        List<MetricInfo> metricInfoList = entityBacklogs.get(entity);
+        String nominalTimeStr = DATE_FORMAT.get().format(nominalTime);
+        MetricInfo metricInfo = new MetricInfo(nominalTimeStr, clusterName);
+        if (!metricInfoList.contains(metricInfo)) {
+            synchronized (metricInfoList) {
+                backlogMetricStore.addInstance(entityName, clusterName, nominalTime, entityType);
+                metricInfoList.add(metricInfo);
+            }
+        }
+    }
+
+    @Override
+    public String getName() {
+        return this.getClass().getSimpleName();
+    }
+
+    @Override
+    public void init() throws FalconException {
+        initInstances();
+        int emitInterval = Integer.parseInt(StartupProperties.get().getProperty(BACKLOG_METRIC_EMIT_INTERVAL,
+                "60000"));
+        int recheckInterval = Integer.parseInt(StartupProperties.get().getProperty(BACKLOG_METRIC_RECHECK_INTERVAL,
+                "60000"));
+        scheduledThreadPoolExecutor1.scheduleAtFixedRate(new BacklogMetricEmitter(),
+                1, emitInterval, TimeUnit.MILLISECONDS);
+        scheduledThreadPoolExecutor2.scheduleAtFixedRate(new BacklogCheckService(),
+                1, recheckInterval, TimeUnit.MILLISECONDS);
+    }
+
+    private void initInstances() throws FalconException {
```

Comment:
If I'm not wrong, this service will run on Falcon Server and not Prism. In that case, shouldn't this server only initialize o nly with Entities that are scheduled on that cluster, and not all entities?